trieste/acquisition/function/function.py

```diff
@@ -583,6 +583,217 @@ def _update_expected_improvement_fn(
             self._expected_improvement_fn.update(eta)  # type: ignore
 
 
+class MonteCarloExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
```

Comment:
see my comment below....
```suggestion
class MonteCarloExpectedImprovement(SingleModelAcquisitionBuilder[HasReparamSampler]):
```